client/src/components/EventItem.js

```diff
@@ -137,11 +161,45 @@ function EventItem(props) {
         {
           console.log("Error while trying to cancel attendance");
         }
-      })
+        props.toggleUpdateEvents();
+      }
+    );
+  };
 
-    // Close the modal and update the event list
-    setOpenCancelAttendance(false);
-    props.toggleUpdateEvents();
+  const handlePayment = async (e) => {
+    e.preventDefault()
+
+    setOpenPayment(false);
+
+    const data = {
+      userId: props.currentUser._id,
+      eventId: props.event._id
+    };
```

Comment:
Here, the userId is undefined when I test it on my computer. So it makes the payment process not working